File: serde-generate/runtime/dart/bincode/bincode_deserializer.dart
// Copyright (c) Facebook, Inc. and its affiliates
// SPDX-License-Identifier: MIT OR Apache-2.0

part of bincode;

class BincodeDeserializer extends BinaryDeserializer {
  BincodeDeserializer(Uint8List input) : super(input);

  @override
  int deserializeLength() {
    return deserializeUint64().toInt();
  }

  @override
  int deserializeVariantIndex() {
    return deserializeUint32();
  }

  @override
  void checkThatKeySlicesAreIncreasing(Slice key1, Slice key2) {
    // Not required by the format.
  }
}

File: serde-generate/runtime/kotlin/com/novi/bincode/BincodeDeserializer.kt
// Copyright (c) Zefchain Labs, Inc.
// SPDX-License-Identifier: MIT OR Apache-2.0

package com.novi.bincode

import com.novi.serde.BinaryDeserializer
import com.novi.serde.DeserializationError
import com.novi.serde.Slice

class BincodeDeserializer(input: ByteArray) : BinaryDeserializer(input, Long.MAX_VALUE) {
    @Throws(DeserializationError::class)
    override fun deserialize_f32(): Float {
        return Float.fromBits(getInt())
    }

    @Throws(DeserializationError::class)
    override fun deserialize_f64(): Double {
        return Double.fromBits(getLong())
    }

    @Throws(DeserializationError::class)
    override fun deserialize_len(): Long {
        val value = getLong()
        if (value < 0 || value > Int.MAX_VALUE.toLong()) {
            throw DeserializationError("Incorrect length value")
        }
        return value
    }

    @Throws(DeserializationError::class)
    override fun deserialize_variant_index(): Int {
        return getInt()
    }

    @Throws(DeserializationError::class)
    override fun check_that_key_slices_are_increasing(key1: Slice, key2: Slice) {
        // Not required by the format.
    }
}
